### PR TITLE
Improve coding standard

### PR DESCRIPTION
**Pull request by alcaeus**
This adds a rule requiring arrow functions to our coding standard. In addition to that, I enabled a rule requiring single quotes where possible. Commits are split by rule for a better review experience.

The last commit entails no actual changes, this was left over from before we were specifying a PHP version in the config.

### CODE REVIEW

File: tests/PedantryTest.php [hunk 1]
```diff
@@ -32,10 +32,8 @@ public function testMethodsAreOrderedAlphabeticallyByVisibility($className): voi
 
         $methods = array_filter(
             $methods,
-            function (ReflectionMethod $method) use ($class) {
-                return $method->getDeclaringClass() == $class // Exclude inherited methods
-                    && $method->getFileName() === $class->getFileName(); // Exclude methods inherited from traits
-            },
+            fn (ReflectionMethod $method) => $method->getDeclaringClass() == $class // Exclude inherited methods
+                    && $method->getFileName() === $class->getFileName(),
```
Comment (GromNaN) [inline tests/PedantryTest.php:36]: You lost the 2nd comment
```suggestion
                    && $method->getFileName() === $class->getFileName(), // Exclude methods inherited from traits
```

**Review by GromNaN — APPROVED** [review]
LGTM.

File: tests/PedantryTest.php (hunk 1 above)
Comment (alcaeus) [reply tests/PedantryTest.php]: Lesson learned: don't trust `phpcbf` to preserve your comments.

**Review by jmikola — COMMENTED** [review]
One question but LGTM.

**Review by jmikola — APPROVED** [review]

**Pull request merged**